android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt

```diff
@@ -0,0 +1,187 @@
+package com.solana.mobilewalletadapter.clientlib
+
+import android.net.Uri
+import android.util.Log
+import androidx.activity.result.ActivityResultCaller
+import com.solana.mobilewalletadapter.clientlib.protocol.JsonRpc20Client
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
+import com.solana.mobilewalletadapter.clientlib.scenario.LocalAssociationScenario
+import com.solana.mobilewalletadapter.clientlib.scenario.Scenario
+import com.solana.mobilewalletadapter.clientlib.scenario.ScenarioChooser
+import com.solana.mobilewalletadapter.clientlib.scenario.ScenarioTypes
+import com.solana.mobilewalletadapter.common.ProtocolContract
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.IOException
+import java.util.concurrent.CancellationException
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+fun LocalAssociationScenario.begin(): MobileWalletAdapterClient {
+    return start().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+}
+
+fun LocalAssociationScenario.end() {
+    close().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+}
+
+@Suppress("BlockingMethodInNonBlockingContext")
+class MobileWalletAdapter(
+    private val resultCaller: ActivityResultCaller,
+    private val scenarioType: ScenarioTypes = ScenarioTypes.Local,
+    private val timeout: Int = Scenario.DEFAULT_CLIENT_TIMEOUT_MS,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) {
+    private val contract = LocalAssociationIntentContract()
+    private val resultLauncher = resultCaller.registerForActivityResult(contract) { }
+
+    private val scenarioChooser = ScenarioChooser()
+
+    suspend fun authorize(identityUri: Uri, iconUri: Uri, identityName: String): Boolean {
+        return withScenario(associate()) { client ->
```

Comment:
Here and below - if you want to call several MWA methods back-to-back, this is going to continuously switch to and return from the wallet. In addition to the one-off methods defined ehre, what about either a Builder object, to create a chain of requests, or a closure provided to a `run` method that allows chaining together multiple calls?

Reply:
Yes, I was actually thinking about that! I will do that as part of this PR. With a few small changes I think I can make something nice.